doc/rst/technotes/interfaces.rst

```diff
@@ -0,0 +1,235 @@
+.. _readme-interfaces:
+
+.. default-domain:: chpl
+
+
+========================================
+Interface in the Chapel Standard Library
+========================================
+
+Interfaces in Chapel are similar to a feature of the same name in object-oriented
+languages such as Java, and to `traits in Rust <https://doc.rust-lang.org/book/ch10-02-traits.html>`_.
+They provide a way to mark a type as exhibiting certain behavior or implementing
+certain functionality. For instance, a ``hashable`` interface marks a type as
+specifying a ``hash`` method (which would allow it to be used in
+:ref:`associative domains <primers-associative>` or sets). The Chapel standard
```

Comment:
Maybe mention the set module here when mentioning sets?

Reply:
I think it gets a little bit too noisy in the introductory sentence (especially for a parenthesizes segment). There's another link to the `Set` module further down, so I don't feel too bothered by this.